File: Myecoprint_admin/ncej_admin/lib/data/module/type.dart
class Type {
  int typeId;
  String typeArabic;
  String typeEnglish;
  DateTime? createdAt;
  DateTime? updatedAt;
  DateTime? deletedAt;

  Type({
    required this.typeId,
    required this.typeArabic,
    required this.typeEnglish,
  
  });

  factory Type.fromJson(Map<String, dynamic> json) {
    return Type(
      typeId: json['type_id']as int? ??0,
      typeArabic: json['type_arabic']as String? ?? '',
      typeEnglish: json['type_english']as String? ?? '',

    );
  }

  Map<String, dynamic> toJson() {
    return {
      'type_id': typeId,
      'type_arabic': typeArabic,
      'type_english': typeEnglish,
     
    };
  }
}

File: backend_Myecoprint/backend_myecoprint/models/type.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Store= require('./store');
const Type = sequelize.define('Type', {
    type_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  type_arabic: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  type_english: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  updatedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  deletedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
}, {
    timestamps: false, 
    paranoid: true, 
    tableName: 'types', 
  });
  

module.exports = Type;




